scenario-api/src/main/scala/pl/touk/nussknacker/engine/canonize/MaybeArtificial.scala

```diff
@@ -45,15 +45,24 @@ private[engine] object MaybeArtificial {
   // we need to make sure it's unique to prevent weird errors
   private def generateArtificialName() = s"$DummyObjectNamePrefix-${UUID.randomUUID()}"
 
-  def artificialSink(errors: ProcessUncanonizationError*): MaybeArtificial[node.SubsequentNode] =
+  def missingSinkError(errors: ProcessUncanonizationError*): MaybeArtificial[node.SubsequentNode] =
     new MaybeArtificial(
       node.EndingNode(node.Sink(generateArtificialName(), SinkRef(artificalSourceSinkRef, Nil), None)),
       errors.toList
     )
 
   def artificialSource(errors: ProcessUncanonizationError*): MaybeArtificial[node.SourceNode] =
-    artificialSink(errors: _*).map(
+    missingSinkError(errors: _*).map(
       node.SourceNode(node.Source(generateArtificialName(), SourceRef(artificalSourceSinkRef, Nil)), _)
     )
 
+  def addedArtificialDeadEndSink(previousNodeId: String): MaybeArtificial[node.SubsequentNode] =
```

Comment:
Why do you have to create some artifical sink? For filter's false we never created any artificial dead-end and everything was fine

Reply:
If you want to fade out some feature, the code should look like this feature didn't exist any time. If you want to still keep backward compatibility for some period of time, the code keeping this compatibility should be encapsulated and it should be as easy as possible to remove this support. It is our case - we want to remove this validation at all eventually. So let's remove the restriction everywhere where it exists, and after that let's add a little class providing backward compatibility, checking what is the type of ending node and returning the previous error.

Reply:
I created `trait MissingSinkHandler`, with 2 implementations - for the old behavior of returning error, and the new behavior of returning None as the "next" node. It is used as implicit in the ProcessCanonizer

Reply:
Nice